packages/cactus-plugin-keychain-azure-kv/src/test/typescript/mock/plugin-keychain-azure-kv-mock.ts

```diff
@@ -0,0 +1,135 @@
+/**
+ * This class mocks the SecretClient class of "@azure/keyvault-secrets" library
+ * by overriding the SecretClient class methods and storing the secrets In-Memory
+ * TO DO: This class shall be replaced with actual usage of SecretClient class in
+ * the main class located at packages/cactus-plugin-keychain-azure-kv/src/main/typescript/plugin-keychain-azure-kv.ts
+ */
+import {
+  Logger,
+  Checks,
+  LogLevelDesc,
+  LoggerProvider,
+} from "@hyperledger/cactus-common";
+
+import {
+  SecretClient,
+  KeyVaultSecret,
+  SecretProperties,
+  PollerLike,
+  PollOperationState,
+  DeletedSecret,
+  PagedAsyncIterableIterator,
+} from "@azure/keyvault-secrets";
+
+export interface ISecretClientMock {
+  azureKvUrl: string;
+  logLevel?: LogLevelDesc;
+}
+
+export class SecretClientMock implements SecretClient {
+  public static readonly CLASS_NAME = "SecretClientMock";
+  private readonly log: Logger;
+  readonly vaultUrl: string;
+  private readonly secrets: Map<string, string>;
+
+  public get className(): string {
+    return SecretClientMock.CLASS_NAME;
+  }
+
+  constructor(public readonly options: ISecretClientMock) {
+    const fnTag = `${this.className}#constructor()`;
+    Checks.truthy(options, `${fnTag} arg options`);
+
+    const level = this.options.logLevel || "INFO";
+    const label = this.className;
+    this.vaultUrl = this.options.azureKvUrl;
+    this.secrets = new Map<string, string>();
+    this.log = LoggerProvider.getOrCreate({ level, label });
+  }
+
+  async setSecret(secretName: string, value: string): Promise<KeyVaultSecret> {
+    this.secrets.set(secretName, value);
+    const secretProperties: SecretProperties = {
+      vaultUrl: this.vaultUrl,
+      name: secretName,
+    };
+    const keyVaultSecret: KeyVaultSecret = {
+      properties: secretProperties,
+      name: secretName,
+    };
+    return keyVaultSecret;
+  }
+
+  async beginDeleteSecret(
+    name: string,
+  ): Promise<PollerLike<PollOperationState<DeletedSecret>, DeletedSecret>> {
+    this.secrets.delete(name);
+    return (null as unknown) as PollerLike<
```

Comment:
@jagpreetsinghsasan How difficult would it be to return something non-null that satisfies this `PollerLike` type?

Reply:
I have changed it (will upload soon)
![image](https://user-images.githubusercontent.com/56861721/122004601-3cb12280-cdd2-11eb-8e84-c2e8bbc00909.png)